tools/bumper/cnao_repo_commands.go

```diff
@@ -162,6 +166,53 @@ func (cnaoRepoOps *gitCnaoRepo) isPrAlreadyOpened(proposedPrTitle string) (bool,
 	return false, nil
 }
 
+// collectBumpFile is a wrapper for collectModifiedToTreeList
+func (cnaoRepoOps *gitCnaoRepo) collectBumpFile() ([]*github.TreeEntry, error) {
+	return cnaoRepoOps.collectModifiedToTreeList(getAllowedList())
+}
+
+// collectModifiedToTreeList collects the modified files in the allowedList paths and returns a github tree entry list
+func (cnaoRepoOps *gitCnaoRepo) collectModifiedToTreeList(allowedList []string) ([]*github.TreeEntry, error) {
+	w, err := cnaoRepoOps.gitRepo.repo.Worktree()
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to get bumping repo Worktree")
+	}
+
+	status, err := w.Status()
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to get bumping repo status")
+	} else if status.IsClean() {
+		return nil, fmt.Errorf("No files to commit")
+	}
+
+	// Create a tree with what to commit.
+	var entries []*github.TreeEntry
+	for localFile, status := range status {
+		if status.Staging == git.Unmodified && status.Worktree == git.Unmodified {
+			continue
+		}
+
+		fileNameWithPath := filepath.Join(cnaoRepoOps.gitRepo.localDir, localFile)
+		content, err := ioutil.ReadFile(fileNameWithPath)
+		if err != nil {
+			return nil, errors.Wrapf(err, "Failed to read local file %s", fileNameWithPath)
+		}
+
+		if fileInGlobList(localFile, allowedList) {
+			logger.Printf("File added to tree: %s", localFile)
+			entries = append(entries, &github.TreeEntry{Path: github.String(localFile), Type: github.String("blob"), Content: github.String(string(content)), Mode: github.String("100644")})
+		} else {
+			logger.Printf("Skipping file %s, not in allowed list", localFile)
```

Comment:
question: why no error is thrown in this case?

Reply:
I understand why you'd want to add an error in this case.. but I don't think it preferable, let me explain.

one of the reasons I added the allowList is that we may be running the bumper locally while debugging (/some other reason we would prefer  to run it like that) , and naturally in this scenario you wouldn't want to treat the WIP bumper files as part of the "bump script" changes. this is where allowList is handy, and in this scenario, you wouldn't want to return error, but only skip it..

In any case the PRs generated need to be reviewed and approved by the moderator

Reply:
oh I understand. Make sense. Thank you for explaining.